finality-aleph/src/network/gossip/service.rs

```diff
@@ -361,21 +403,30 @@ impl<N: RawNetwork, D: Data> Service<N, D> {
         loop {
             tokio::select! {
                 maybe_event = events_from_network.next_event() => match maybe_event {
-                    Some(event) => if let Err(e) = self.handle_network_event(event) {
-                        error!(target: "aleph-network", "Cannot forward messages to user: {:?}", e);
+                    Some(event) => if self.handle_network_event(event).is_err() {
+                        error!(target: "aleph-network", "Cannot forward messages to user.");
                         return;
                     },
                     None => {
                         error!(target: "aleph-network", "Network event stream ended.");
                         return;
                     }
                 },
-                maybe_message = self.messages_from_user.next() => match maybe_message {
-                    Some(Command::Broadcast(message, protocol)) => self.broadcast(message, protocol),
-                    Some(Command::SendToRandom(message, peer_ids, protocol)) => self.send_to_random(message, peer_ids, protocol),
-                    Some(Command::Send(message, peer_id, protocol)) => self.send(message, peer_id, protocol),
+                maybe_message = self.messages_from_authentication_user.next() => match maybe_message {
```

Comment:
nit: divide it into two `match` statements instead of repeating `Some`.

Reply:
chatgpt says: Using a nested match expression is also a valid approach, but it may lead to more verbose code and potentially make it harder to read and understand the logic. Additionally, in this case, using nested matches may not be necessary because the cases being handled are not particularly complex.

oh, and also @timorl said something similar ;) let's leave it that way, shall we?

Reply:
I hate that ChatGPT agrees with me. D'x

Reply:
I didn't mean nested....

Reply:
```suggestion
	maybe_message = self.messages_from_authentication_user.next() => {
		let message = match maybe_message {
			Some(msg) => msg,
			None => {
				error!(target: "aleph-network", "Authentication user message stream ended.");
				return;
			},
		};
		match message {
			Command::Broadcast(message) => self.broadcast_authentication(message),
			Command::SendToRandom(message, peer_ids) => self.send_to_random_authentication(message, peer_ids),
			Command::Send(message, peer_id) => self.send_authentication_data(message, peer_id),
		}
	}
```

Reply:
haha super ugly. Move those to separate functions, return `Result` and use `?` operator, etc.